### PR TITLE
fixed uniswap permit 2 dependency.

### PR DESCRIPTION
**Pull request by jaysinghdhakad**
Screenshot from 2024-11-01 16-55-20

### CODE REVIEW

File: meta-aggregator-contract/contracts/MetaAggregatorSwapContract.sol [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+            revert AmountInMustBeGreaterThanZero();
+        }
+        if (minAmountOut == 0) {
+            revert MinAmountOutMustBeGreaterThanZero();
+        }
+        if (address(tokenIn) == address(tokenOut)) {
+            revert TokenInAndTokenOutCannotBeSame();
+        }
+
+        TransferHelper.safeApprove(address(tokenIn), aggregator, 0);
```
Comment (langnavina97) [inline meta-aggregator-contract/contracts/MetaAggregatorSwapContract.sol:209]: If we deploy to ETH we should add an if check here and only reset the approval for USDT

**Comment by langnavina97** [comment]
I think it makes more sense to add the contracts to the contracts folder and add the test cases to the existing ones

**Comment by aj07** [comment]
> I think it makes more sense to add the contracts to the contracts folder and add the test cases to the existing ones

@jaysinghdhakad I think this is correct.. Lets make this change quickly and send to auditor..

File: contracts/meta-aggregator/MetaAggregatorManager.sol [hunk 2]
```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import {ReentrancyGuard} from "@openzeppelin/contracts-5.0.2/utils/ReentrancyGuard.sol";
```
Comment (langnavina97) [inline contracts/meta-aggregator/MetaAggregatorManager.sol:4]: Any reason you're using a different open zeppelin version here?

Comment (jaysinghdhakad) [reply contracts/meta-aggregator/MetaAggregatorManager.sol:4]: I used the  one which I had in my repository for the contracts to be the way I developed.

Comment (langnavina97) [reply contracts/meta-aggregator/MetaAggregatorManager.sol:4]: Can you please use the same version we're using everywhere in the repo?

Comment (jaysinghdhakad) [reply contracts/meta-aggregator/MetaAggregatorManager.sol:4]: The one I am using is new and we are supposed to use lattest verion of openzeppline

File: contracts/meta-aggregator/MetaAggregatorManager.sol [hunk 3]
```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
```
Comment (langnavina97) [inline contracts/meta-aggregator/MetaAggregatorManager.sol:2]: We're using `pragma solidity 0.8.17;` everywhere in our contracts, please also use this version in your contracts

File: meta-aggregator-contract/contracts/MetaAggregatorSwapContract.sol (hunk 1 above)
Comment (langnavina97) [reply meta-aggregator-contract/contracts/MetaAggregatorSwapContract.sol]: @aj07 @Havoc19 what do you think? do we need to add the check here?

**Review by langnavina97 — APPROVED** [review]

**Pull request merged**